include/sparrow/contracts.hpp

```diff
@@ -12,92 +12,149 @@
-#    error "std::format usage is requested but not available"
-#  endif
+#    if defined(SPARROW_CONTRACTS_USE_STD_FORMAT) and not defined(__cpp_lib_format)
+#        error "std::format usage is requested but not available"
+#    endif
 #endif
 
 // user specified to use neither std::format nor std::print, but C's formatting
 #if defined(SPARROW_CONTRACTS_USE_CFORMAT) && SPARROW_CONTRACTS_USE_CFORMAT == 1
-# ifdef SPARROW_CONTRACTS_USE_STD_FORMAT
-#   undef SPARROW_CONTRACTS_USE_STD_FORMAT
-# endif
-# ifdef SPARROW_CONTRACTS_USE_STD_PRINT
-#   undef SPARROW_CONTRACTS_USE_STD_PRINT
-# endif
+#    ifdef SPARROW_CONTRACTS_USE_STD_FORMAT
+#        undef SPARROW_CONTRACTS_USE_STD_FORMAT
+#    endif
+#    ifdef SPARROW_CONTRACTS_USE_STD_PRINT
+#        undef SPARROW_CONTRACTS_USE_STD_PRINT
+#    endif
+#endif
+
+#if defined(__GNUC__)
+#    define SPARROW_CONTRACTS_IGNORE_WARNINGS \
+      _Pragma("GCC diagnostic push") \
+      _Pragma("GCC diagnostic ignored \"-Wall\"") \
+      _Pragma("GCC diagnostic ignored \"-Wformat-security\"")
+#    define SPARROW_CONTRACTS_RESTORE_WARNINGS \
+      _Pragma("GCC diagnostic pop")
+#elif defined(__clang__)
+#    define SPARROW_CONTRACTS_IGNORE_WARNINGS \
+      _Pragma("clang diagnostic push") \
+      _Pragma("clang diagnostic ignored \"-Weverything\"")
+#    define SPARROW_CONTRACTS_RESTORE_WARNINGS \
+      _Pragma("clang diagnostic pop")
+#elif defined(_MSC_VER)
+#    define SPARROW_CONTRACTS_IGNORE_WARNINGS \
+      _Pragma("warning(push)") \
+      _Pragma("warning(disable : 4774)") // 'var' has different type in 'file1' and 'file2': 'type1' and 'type2'
+#    define SPARROW_CONTRACTS_RESTORE_WARNINGS \
+      _Pragma("warning(pop)")
+#else
+#    define SPARROW_CONTRACTS_IGNORE_WARNINGS
+#    define SPARROW_CONTRACTS_RESTORE_WARNINGS
 #endif
 
 #ifndef SPARROW_CONTRACTS_LOG_FAILURE
-# if defined(SPARROW_CONTRACTS_USE_STD_PRINT) && SPARROW_CONTRACTS_USE_STD_PRINT == 1
-#   include <print>
-#   include <cstdio>
-#   define SPARROW_CONTRACTS_LOG_FAILURE( expr__, message__ ) \
-      ::std::print(stderr, "Assertion Failed ({}:{}): {} - ({} is wrong)\n", __FILE__, __LINE__, message__, #expr__ )
-# elif defined(SPARROW_CONTRACTS_USE_STD_FORMAT) && SPARROW_CONTRACTS_USE_STD_FORMAT == 1
-#   include <format>
-#   include <cstdio>
-#   define SPARROW_CONTRACTS_LOG_FAILURE( expr__, message__ ) \
-      ::fprintf(stderr, ::std::format("Assertion Failed ({}:{}): {} - ({} is wrong)\n", __FILE__, __LINE__, message__, #expr__ ).c_str())
-# else
-#   include <cstdlib>
-#   include <cstdio>
-#   define SPARROW_CONTRACTS_LOG_FAILURE( expr__, message__ ) \
-      ::fprintf(stderr, "Assertion Failed (%s:%i): %s - (%s is wrong)\n", __FILE__, __LINE__, message__, #expr__ )
-# endif
+#    if defined(SPARROW_CONTRACTS_USE_STD_PRINT) && SPARROW_CONTRACTS_USE_STD_PRINT == 1
+#        include <cstdio>
+#        include <print>
+
+#        define SPARROW_CONTRACTS_LOG_FAILURE(expr__, message__)  \
```

Comment:
I think the formatting is a bit extreme for these function calls 😂 
Can you deactivate it for the 3 print calls and if you want to indent it in 2 lines it's fine? Right now it's header to read than the original.

Reply:
Maybe also deactivate the formatting around the `do/while` block 🥲

Reply:
Oups, sorry I didn't view this auto indent, I will add a NOLINT block